tests/rptest/tests/admin_uuid_operations_test.py

```diff
@@ -82,18 +89,211 @@ def _uuids_updated():
+        )
+        current_uuid = self.scrape_uuid(to_stop)
+        assert current_uuid is not None, "Didn't find UUID in logs"
+
+        self.logger.debug("Restart the node again (but keep the disk)")
+
+        THE_OVERRIDE = f"{current_uuid} -> ID: '{initial_to_stop_id}' ; UUID: '{old_uuid}'"
+        if mode == "cfg-override":
+            self.logger.debug(
+                f"Override with known-good uuid/id via node config: {THE_OVERRIDE}"
+            )
+            self._restart_node(
+                to_stop,
+                dict(node_id_overrides=[
+                    dict(current_uuid=current_uuid,
+                         new_uuid=old_uuid,
+                         new_id=initial_to_stop_id)
+                ], ),
+                drop_disk=False,
+            )
+        elif mode == "no-override":
+            self.logger.debug(
+                "Omit the override to confirm that we're still stuck in that case"
+            )
+            with expect_exception(TimeoutError, lambda _: True):
+                self._restart_node(to_stop, drop_disk=False)
+            self.logger.debug("And short circuit the test case")
+            return
+        else:
+            assert False, f"Unexpected mode: '{mode}'"
+
+        self.logger.debug(
+            "Wait until the target node reflects the given overrides")
+
+        wait_until(lambda: self.admin.get_broker_uuid(to_stop)['node_id'] ==
+                   initial_to_stop_id,
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg=f"{to_stop.name} did not take the ID override")
+
+        wait_until(lambda: self.admin.get_broker_uuid(to_stop)['node_uuid'] ==
+                   old_uuid,
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg=f"{to_stop.name} did not take the UUID override")
+
+        self.logger.debug(f"Decommission ghost node [{ghost_node_id}]...")
+        self.admin.decommission_broker(ghost_node_id, node=to_stop)
+
+        self.logger.debug(f"...and wait for the cluster to become healthy")
+
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=30,
+                                 backoff_sec=1)
+
+        self.logger.debug(
+            "Check that all this state sticks across a rolling restart")
+
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            auto_assign_node_id=True)
+
+        after_restart = self.admin.get_broker_uuid(to_stop)
+
+        assert after_restart[
+            'node_id'] == initial_to_stop_id, f"Bad node id after override: {after_restart['node_id']}"
+        assert after_restart[
+            'node_uuid'] == old_uuid, f"Bad node uuid after override: {after_restart['node_uuid']}"
+
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=30,
+                                 backoff_sec=1)
```

Comment:
I think this will be flaky because `get_broker_uuid` may fail if there's no controller leader immediately after the restart. You could swap these two checks, first wait for healthy and then assert on `get_broker_uuid`.

Reply:
I'm a bit fuzzy on whether a controller leader is strictly required for this check to pass, but point taken. It's better placed after the health check - and I think even better if we direct the request to some other node and wrap in a waiter?